### DISCUSSION TITLE
Can I declare a setup file for a specific test or test file?

### DISCUSSION THREAD

**Discussion by CodyBontecou** [vitest-dev/vitest#5673] [Q&A] [answered]
I have the need to utilize different MSW server's depending on the test that is running. 
I'm currently utilizing a setupFile like so:

```ts
import { errorServer } from '@/mocks/msw/server'
import { afterAll, afterEach, beforeAll, vi } from 'vitest'

beforeAll(() => {
  // Mocking new Date()
  vi.useFakeTimers()
  vi.setSystemTime(new Date('2024-05-03T11:00:00Z'))

  // Establish API mocking before all tests.
  errorServer.listen()
})
afterEach(() => errorServer.resetHandlers())
afterAll(() => {
  vi.useRealTimers()
  errorServer.close()
})
```

But I only want this particular file when testing against error states and have a separate setup file for non-error-state-related tests.

**Comment by hi-ogawa** [comment, marked as answer]
You can extract `beforeAll` etc... as a utility and ditch `setupFiles` completely

```ts
// helper.ts
import { errorServer } from '@/mocks/msw/server'
import { afterAll, afterEach, beforeAll, vi } from 'vitest'

export function setupErrorServer() {
  beforeAll(() => ... );
  afterEach(() => ...);
  afterAll(() => ...);
}

// some-file.test.ts
import { test } "vitest"
import { setupErrorServer } from "./helper"

setupErrorServer();

test("something", () => {
  ...
})
```

If you want to do Vitest magic, then you could technically check current file inside `setupFiles`:

```ts
// my-setup.ts
beforeAll(ctx => { ctx.file; })
```

or you could use Vitest workspace https://vitest.dev/guide/workspace.html to have a completely separate `setupFiles`, but either seem overkill.

**Reply by CodyBontecou** [reply]
Very nice, thanks for the tip. This works perfect.
